src/builder.rs

```diff
@@ -110,15 +112,22 @@ impl<K: EnrKey> EnrBuilder<K> {
 
     /// Generates the rlp-encoded form of the ENR specified by the builder config.
     fn rlp_content(&self) -> BytesMut {
-        let mut stream = RlpStream::new_with_buffer(BytesMut::with_capacity(MAX_ENR_SIZE));
-        stream.begin_list(self.content.len() * 2 + 1);
-        stream.append(&self.seq);
+        let mut list = Vec::<u8>::with_capacity(MAX_ENR_SIZE);
+        list.append(&mut alloy_rlp::encode(self.seq));
         for (k, v) in &self.content {
-            stream.append(k);
-            // The values are stored as raw RLP encoded bytes
-            stream.append_raw(v, 1);
+            // Keys are bytes
+            list.append(&mut alloy_rlp::encode(k.as_slice()));
```

Comment:
```suggestion
            k.as_slice().encode(&mut list);
```
note that this is not just a style thing:
`alloy_rlp::encode` allocates a vector, and `append` drains that new vector onto itself. This on the other hand, writes directly to the already allocated vector. We have saved ourselves a temporary vec allocation.

Also, `alloy_rlp::encode` potentially calls the default implementation of `Encodable::length` which first encodes and then gets the length of the encoded data. This means that calling `alloy_rlp::encode` potentially encodes the data twice, first to get the size of the encoded data and allocate a vector to store it, and then actually encode it. Calling the `Encodable::encode` method directly ensure we use the optimized versions of `length` in all cases